.github/workflows/github-actions.yml

```diff
@@ -0,0 +1,41 @@
+name: GitHub Actions
+on: [push]
+
+jobs:
+  test:
+
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        python-version: ["3.8"]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      # You can test your matrix by printing the current Python version
+      - name: Display Python version
+        run: python -c "import sys; print(sys.version)"
+      - name: Install Netzob
+        run: |
+          ls ${{ github.workspace }}
+          sudo apt-get install -y python3 python3-dev python3-setuptools virtualenv build-essential libpcap-dev libgraph-easy-perl libffi-dev
+          cd netzob/
+          mkdir venv
+          virtualenv venv
+          source venv/bin/activate
+          pip3 install Cython==0.29.32
+          pip3 install -e .
+      - name: Run test
+        run: |
+          cd netzob/
+          source venv/bin/activate
+          python3 setup.py test
+      - name: Generate documentation
+        run: |
+          cd netzob/
+          source venv/bin/activate
+          sphinx-build -b html doc/documentation/source/ doc/documentation/build/
```

Comment:
This section could be moved to a dedicated GitHub Actions workflow that could be triggered on `push` on the “main” branch. It could also be possible to push the documentation on GitHub Pages.

Reply:
issue created